### PR TITLE
Simplify Fourier-Bessel basis list of zeros

### LINKED ISSUE #609: MATLAB hold-overs in Fourier Bessel zero finding method
The following should be modified to be a list of arrays (possibly of variable lengths, for each ell), rather than a zero padded NumPy array.

cf. #599 

https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/6cb389edf21e811224cc98f4fde0d63ed778b9a7/src/aspire/basis/basis.py#L79-L85

### PR DESCRIPTION
**Pull request by chris-langfield**
This will close out #609 

It turned out the `m_reshape` was totally unnecessary in the first place: you could achieve the same effect by doing `np.array(zeros).T`!

This test turns `basis.r0` from an array of shape `(max_num_zeros, ell_max)` (i.e., the first dimension tracks the zeros of a given Bessel function, while the second tracks the Bessel order) to a list of length `ell_max`, where the i'th element is a 1d numpy array containing exactly as many zeros as we have for Bessel order i. That is, it is a ragged list.

### CODE REVIEW

**Review by j-c-c — APPROVED** [review]
LGTM!

File: src/aspire/basis/fb.py [hunk 1]
```diff
@@ -47,10 +46,6 @@ def _calc_k_max(self):
...
         self.k_max = np.array(n, dtype=int)
 
-        max_num_zeros = max(len(z) for z in zeros)
-        for i, z in enumerate(zeros):
-            zeros[i] = np.hstack(
-                (z, np.zeros(max_num_zeros - len(z), dtype=self.dtype))
-            )
-
-        self.r0 = m_reshape(np.hstack(zeros), (-1, self.ell_max + 1)).astype(self.dtype)
+        # set the zeros for each ell
```
Comment (garrettwrong) [inline src/aspire/basis/fb.py:49]: Thanks for the comment.

File: src/aspire/basis/fb.py [hunk 2]
```diff
@@ -47,10 +46,6 @@ def _calc_k_max(self):
...
+        # this is a ragged list of 1d ndarrays, where the i'th element is of size self.k_max[i]
+        self.r0 = zeros
```
Comment (garrettwrong) [inline src/aspire/basis/fb.py:51]: Laughed at the deletion-replacement here. Nice. Thanks

**Review by garrettwrong — APPROVED** [review]
LGTM thanks!!

**Pull request merged**